azext_iot/operations/events3/_events.py

```diff
@@ -125,7 +127,12 @@ def _get_conn_props():
     def _output_msg_kpi(msg):
         # TODO: Determine if amqp filters can support boolean operators for multiple conditions
         origin = str(msg.annotations.get(b'iothub-connection-device-id'), 'utf8')
-        if device_id and origin != device_id:
+        if device_id:
```

Comment:
Do we have to use regex on every request?  I would rather us not take the regex overhead (even if small) if we don't have to.